app/assets/javascripts/task_list.coffee

```diff
@@ -94,145 +87,174 @@
-      if line.match(endFencesPattern)
-        # Stop ignoring lines once the code block is closed.
-        inCodeBlock = false
-    else if line.match(startFencesPattern)
-      # Start ignoring lines inside a code block.
-      inCodeBlock = true
-    else if line in clean && line.match(itemPattern)
-      index += 1
-      if index == itemIndex
-        line =
-          if checked
-            line.replace(incompletePattern, complete)
-          else
-            line.replace(completePattern, incomplete)
-    line
-  result.join("\n")
-
-# Updates the $field value to reflect the state of $item.
-# Triggers the `tasklist:change` event before the value has changed, and fires
-# a `tasklist:changed` event once the value has changed.
-updateTaskList = ($item) ->
-  $container = $item.closest '.js-task-list-container'
-  $field     = $container.find '.js-task-list-field'
-  index      = 1 + $container.find('.task-list-item-checkbox').index($item)
-  checked    = $item.prop 'checked'
-
-  event = $.Event 'tasklist:change'
-  $field.trigger event, [index, checked]
-
-  unless event.isDefaultPrevented()
-    $field.val updateTaskListItem($field.val(), index, checked)
-    $field.trigger 'change'
-    $field.trigger 'tasklist:changed', [index, checked]
-
-# When the task list item checkbox is updated, submit the change
-$(document).on 'change', '.task-list-item-checkbox', ->
-  updateTaskList $(this)
-
-# Enables TaskList item changes.
-enableTaskList = ($container) ->
-  if $container.find('.js-task-list-field').length > 0
-    $container.
-      find('.task-list-item').addClass('enabled').
-      find('.task-list-item-checkbox').attr('disabled', null)
-    $container.addClass('is-task-list-enabled').
-      trigger 'tasklist:enabled'
-
-# Enables a collection of TaskList containers.
-enableTaskLists = ($containers) ->
-  for container in $containers
-    enableTaskList $(container)
-
-# Disable TaskList item changes.
-disableTaskList = ($container) ->
-  $container.
-    find('.task-list-item').removeClass('enabled').
-    find('.task-list-item-checkbox').attr('disabled', 'disabled')
-  $container.removeClass('is-task-list-enabled').
-    trigger 'tasklist:disabled'
-
-# Disables a collection of TaskList containers.
-disableTaskLists = ($containers) ->
-  for container in $containers
-    disableTaskList $(container)
-
-$.fn.taskList = (method) ->
-  $container = $(this).closest('.js-task-list-container')
-
-  methods =
-    enable: enableTaskLists
-    disable: disableTaskLists
-
-  methods[method || 'enable']($container)
+NodeArray = (nodeList) -> Array.prototype.slice.apply(nodeList)
+
+closest = (el, className) ->
+  while el && !el.classList.contains className
+    el = el.parentNode
+  el
+
+createEvent = (eventName, detail) ->
+  if typeof Event == 'function'
+    event = new Event eventName, {bubbles: true, cancelable: true}
+    event.detail = detail
+  else
+    event = document.createEvent 'CustomEvent'
```

Comment:
custom event needs a polyfill. Maybe add that as a dependency in the documentation?